compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -1384,7 +1385,7 @@ class CheckCaptures extends Recheck, SymTransformer:
               if boxErrors != null then boxErrors += msg
               if ctx.settings.YccDebug.value then
                 println(i"cannot box/unbox $actual vs $expected")
-              return actual
+              boundary.break(actual)
```

Comment:
if you enable tracing then this return starts to warn, due to the surrounding trace call so this is here